src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -149,16 +149,18 @@ private Expression ParseLambdaExpression()
 
 				var startExpr = _parsePosition;
 
+				// decide where the lambda expression body ends:
+				// it's the first unmatched parenthesis, curly bracket, of comma
```

Comment:
Would this "succeed" on the following code?

`courseList.Select(x => new PageType() { PageName = x.PageName, VisualCount = 5 ))`

Reply:
Yes, the matching will "work", but this algorithm is only here to detect where the body of a lambda expression ends, so that we know which string to parse. That string is then parsed independently.

Here the lambda expression body will be considered to be `x => new PageType() { PageName = x.PageName, VisualCount = 5 )`, but that's not a valid expression, and therefore the parsing will fail.

Reply:
yes

Reply:
Yes, sorry!